template/hooks/make/run-make

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+# shellcheck source=../git_config_wrapper.sh
+. "$GIT_DIR/hooks/git_config_wrapper.sh"
+
+get_hook_config make target target optional test
+
+if ! make --file="$(git rev-parse --show-toplevel)/Makefile" $target
+then
+	exit 1
```

Comment:
You might want to skip the `if` and do something like that:

```bash
make $target
exit $? # $? contains the status code of the last command
```

This way you have the right status code corresponding to the last command run in the makefile

But I'm not sure if it will break other hooks or not (depends on how git-hooks are run internally)

Reply:
Yeah not sure either, but I guess I can use `$?` instead of `1` anyway, that's what I did in some other hooks it seems

Reply:
I think that's equivalent to just `make --file="$(git rev-parse --show-toplevel)/Makefile" $target`

Reply:
As a matter of fact

```bash
make $target
exit $?
```

and

```bash
make -C "$(git rev-parse --show-toplevel)" $target
```

are only identical if the script calling the file run it using a sub-process. This will behave differently if you `source` the file because on the former case it will stop even if the script succeed (which is not what you want).